src/mmrelay/matrix/events.py

```
@@ -0,0 +1,766 @@
+                )
+                return
+            facade.logger.warning(
+                f"Key request attempt {attempt + 1} failed for event {event.event_id}, retrying..."
+            )
+            facade.logger.debug("Key request failure details", exc_info=True)
+
+        if backoff_delay is not None:
+            await asyncio.sleep(backoff_delay)
+
+
+async def on_room_message(
```

Ruff (0.15.9)

[warning] Too many return statements (21 > 6)

(PLR0911)